event_registration_multi_qty/models/event.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Sergio Teruel<sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class EventEvent(models.Model):
+    _inherit = 'event.event'
+
+    registration_multi_qty = fields.Boolean(
+        string='Allow multiple attendees per registration',
+    )
+
+    @api.multi
+    @api.depends('seats_max', 'registration_ids.state',
+                 'registration_ids.qty')
+    def _compute_seats(self):
+        for event in self:
+            if not event.registration_multi_qty:
+                return super(EventEvent, self)._compute_seats()
+            event.seats_unconfirmed = event.seats_reserved = 0
+            event.seats_used = event.seats_available = 0
+            registrations = self.env['event.registration'].read_group([
+                ('event_id', '=', event.id),
+                ('state', 'in', ['draft', 'open', 'done'])
+            ], ['state', 'qty'], ['state'])
+            for registration in registrations:
+                if registration['state'] == 'draft':
+                    event.seats_unconfirmed += registration['qty']
+                elif registration['state'] == 'open':
+                    event.seats_reserved += registration['qty']
+                elif registration['state'] == 'done':
+                    event.seats_used += registration['qty']
+            if event.seats_max > 0:
+                event.seats_available = event.seats_max - (
+                    event.seats_reserved + event.seats_used)
+            event.seats_expected = (
+                event.seats_unconfirmed + event.seats_reserved +
+                event.seats_used)
+
+    @api.multi
+    @api.constrains('registration_multi_qty')
+    def _check_attendees_qty(self):
+        for event in self:
+            if not event.registration_multi_qty and \
+                    max(event.registration_ids.mapped('qty') or [0]) > 1:
+                raise ValidationError(
+                    _('You can not inactive this option if there is '
+                      'registrations with quantities grather than one.'))
+
+
+class EventRegistration(models.Model):
+    _inherit = 'event.registration'
+
+    qty = fields.Integer(
+        string='Quantity',
+        required=True,
+        default=1,
+    )
+
+    @api.multi
+    @api.constrains('qty')
+    def _check_attendees_qty(self):
+        for registration in self:
+            if not registration.event_id.registration_multi_qty and \
+                    max(registration.event_id.registration_ids.mapped(
+                        'qty') or [0]) > 1:
```

Comment:
This check is IMHO overloaded. While editing a registration, you don't need to check all others from the same event, since they don't changed.

`if True in self.mapped("event_id.registration_multi_qty") and self.filtered(lambda reg: reg.qty != 1)` should be enough